### DISCUSSION TITLE
What do you think about making Torch fully available in Power Puter?

### DISCUSSION THREAD

**Discussion by Theroniya** [rgthree/rgthree-comfy#639] [Ideas]
We can use many torch features. Indexing, slicing, multiplication, and tensor methods such as mask.unsqueeze(-1).

But some features, such as torch.clamp(), are not available as tensor methods.

By the way, Power Putter is already insecure.  This won't make it any worse.
Connect a tensor to an input and write: a.numpy().tofile(‘/tmp/test’)

**Comment by rgthree** [comment]
Not opposed to it, but with a vetted allowlist. I'm just as familiar with all torch calls, and haven't gone through a process of adding it.

> By the way, Power Putter is already insecure. This won't make it any worse. Connect a tensor to an input and write: a.numpy().tofile(‘/tmp/test’)

Thanks! I agree that saving to the filesystem directly shouldn't be allowed. That's now patched, and I appreciate you finding it.
